src/pip/_internal/cli/base_command.py

```diff
@@ -217,6 +218,11 @@ def _main(self, args):
             logger.critical('Operation cancelled by user')
             logger.debug('Exception information:', exc_info=True)
 
+            return ERROR
+        except NetworkConnectionError as exc:
+            logger.critical(str(exc))
+            logger.debug('Exception information: ', exc_info=True)
+
```

Comment:
Any reason this is not merged into the `except (InstallationError, UninstallationError, BadCommand) as exc:` block? They seem to do exactly the same thing.

Reply:
Ah! no. I didn't get on my mind before. Would update it.

Reply:
However, to me, it seems the flow never comes here. As when we raise the custom exception, it somehow doesn't bubble up to this stage. Do we need to catch and raise the exception on each function call that may lead to the `pip._internal.network.utils.response_chunks()`?

Reply:
I could think of this, as "Exception information" doesn't show up in the error log.

Reply:
Ah! My bad, it does show up when I check the verbose output.